omnipaxos/src/sequence_paxos/leader.rs

```diff
@@ -339,136 +293,35 @@ where
-                        // manually rollback snapshot and log if append suffix fails
-                        self.internal_storage.rollback_and_panic_if_err(
-                            &accepted_res,
-                            vec![
-                                RollbackValue::AcceptedRound(old_accepted_round),
-                                RollbackValue::DecidedIdx(old_decided_idx),
-                                RollbackValue::Log(old_log_res.unwrap()),
-                                RollbackValue::Snapshot(
-                                    old_compacted_idx,
-                                    old_snapshot_res.unwrap(),
-                                ),
-                            ],
-                            "storage error while trying to write log entries",
-                        );
-                    }
-                    None => {
-                        // no snapshot, only suffix
-                        let result = if max_promise_meta.n_accepted == old_accepted_round {
-                            self.internal_storage
-                                .append_entries_without_batching(suffix)
-                        } else {
-                            self.internal_storage.append_on_decided_prefix(suffix)
-                        };
-                        self.internal_storage.rollback_and_panic_if_err(
-                            &result,
-                            vec![
-                                RollbackValue::AcceptedRound(old_accepted_round),
-                                RollbackValue::DecidedIdx(old_decided_idx),
-                            ],
-                            "storage error while trying to write log entries",
-                        );
-                    }
-                }
-            }
-            None => {
-                // I am the most updated
-                self.append_pending_proposals();
-                self.adopt_pending_stopsign();
-            }
-        }
-        match max_stopsign {
-            Some(ss) => {
-                self.accept_stopsign(ss);
+        let decided_idx = self.leader_state.get_max_decided_idx();
+        let mut new_accepted_idx = self
+            .internal_storage
+            .sync_log(max_promise_meta.n_accepted, decided_idx, max_promise_sync)
+            .expect(WRITE_ERROR_MSG);
+
+        if self.internal_storage.get_stopsign().is_none() {
+            // Adopt pending proposals
+            if !self.pending_proposals.is_empty() {
+                let pending_entries = std::mem::take(&mut self.pending_proposals);
+                new_accepted_idx = self
+                    .internal_storage
+                    .append_entries_without_batching(pending_entries)
+                    .expect(WRITE_ERROR_MSG);
             }
-            None => {
-                self.append_pending_proposals();
-                self.adopt_pending_stopsign();
+            if let Some(ss) = self.pending_stopsign.take() {
```

Comment:
we do a check for `self.internal_storage.get_stopsign().is_none()` in the if statement just before. This makes me wonder:
1. Do we need the `self.pending_stopsign `field at all?
2. Why don't we do the match on pending_stopsign.take() instead of doing two different ifs?

Reply:
right... I think the usage of function and naming is confusing here.

- In the first if statement: instead of using `self.internal_storage.get_stopsign().is_none()`, use the `accepted_reconfiguration()` function instead.
- I think the `pending_stopsign` field should be renamed to `buffered_stopsign`, as pending can be interpreted as in the progress of being decided.